src/python/pants/engine/internals/scheduler.py

```diff
@@ -75,6 +75,13 @@
 Workunit = Dict[str, Any]
 
 
+@dataclass(frozen=True)
+class CurrentGoals:
+    """The currently running goals."""
```

Comment:
```suggestion
    """The currently running goals if any."""
```

It's possible for no goals to be running: probably the intrinsic `@rule` should fail in that case.

Reply:
Why fail? The `.rule_names` could be empty in that case, is that not a valid scenario..?